plugins/woocommerce/src/Admin/API/Reports/Products/DataStore.php

```diff
@@ -432,17 +466,79 @@ public static function sync_order_products( $order_id ) {
 		$decimals       = wc_get_price_decimals();
 		$round_tax      = 'no' === get_option( 'woocommerce_tax_round_at_subtotal' );
 
+		$parent_order_id = $order->get_parent_id();
+
+		$is_full_refund = false;
+		$refund_type    = $order->get_meta( '_refund_type' );
+		// When the order is a full refund, there is no order items.
+		// We need to get the parent order items, and exclude the items that is already being patially refunded.
+		if ( 'shop_order_refund' === $order->get_type() && 'full' === $refund_type ) {
+			$is_full_refund     = true;
+			$parent_order_id    = $order->get_parent_id();
+			$parent_order       = wc_get_order( $parent_order_id );
+			$parent_order_items = $parent_order->get_items();
+
+			// Get the partially refunded product ids.
+			$query = "
+				SELECT product_lookup.product_id
+				FROM {$table_name} AS product_lookup
+				INNER JOIN {$wpdb->prefix}wc_order_stats AS order_stats
+					ON order_stats.order_id = product_lookup.order_id
+				WHERE 1 = 1
+					AND order_stats.parent_id = %d
+					AND order_stats.total_sales < 0
+			";
+
+			$refunded_product_ids = $wpdb->get_results(
+				$wpdb->prepare(
+					// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+					$query,
+					$parent_order_id
+				),
+				ARRAY_A
+			);
+
+			$refunded_product_ids = array_map(
+				function ( $refunded_product_id ) {
+					return $refunded_product_id['product_id'];
+				},
+				$refunded_product_ids
+			);
```

Comment:
If we aren't worried about type casting, we can use `array_column` for this. Or alternatively if we use `wp_list_pluck` then we can work with a list of objects and we don't need to change the type to `ARRAY_A` for the DB call.

```suggestion
			$refunded_product_ids = array_column( $refunded_product_ids, 'product_id' );
```